RecoLocalTracker/SiPhase2VectorHitBuilder/src/VectorHitBuilderAlgorithm.cc

```diff
@@ -0,0 +1,574 @@
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderAlgorithm.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit2D.h"
+
+bool VectorHitBuilderAlgorithm::LocalPositionSort::operator()(Phase2TrackerCluster1DRef clus1,
+                                                              Phase2TrackerCluster1DRef clus2) const {
+  static std::map<std::pair<Phase2TrackerCluster1DRef, Phase2TrackerCluster1DRef>, bool> cache;
```

Comment:
not exactly thread safe

Reply:
may I ask you where is cleaned? (just to avoid a massive memory leak...)

Reply:
All this stuff is removed again.